addons/survey/models/survey_survey.py

```diff
@@ -413,6 +472,8 @@ def next_page_or_question(self, user_input, page_or_question_id):
     def _get_survey_questions(self, answer=None, page_id=None, question_id=None):
         questions, page_or_question_id = None, None
 
+        if answer and answer.is_session_answer:
```

Comment:
I've added a docstring, it's not perfect but it's better than nothing.

IMO that method should not return a tuple, but only the question_ids.
The second tuple parameter is strange since it comes from the method arguments and is only relevant for 2 of 4 handled uses cases.
To clean later?